### PR TITLE
Adds New Changeling Ability: Hive Head

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR adds a new changeling ability to the game, entitled 'Hive Head'. A basic description is shown below, as seen in-game.

image

Hive Head is a two-point ability for Changeling which encases your head in a bee hive. The bee hive is rather unremarkable as a defensive utility, boasting extremely poor resistances to all damage types except for BIO and providing flash resistance (your eyes are being covered). Along with this, since Hive Head covers your head, you cannot use it conjunction with chitinous armor, since it requires your head slot be open. While this all sounds rather bad, the major reason to use this ability is upon activation, you gain another ability which costs no chemicals and operates on a cooldown: Release Bees.

Release Bees summons 6 bees on your location which will attack anything that isn't you or directly allied to you for 20 seconds, which afterwards the bees will vanish. By default, these bees are standard and do not inject any sort of chemicals into your victims, so while they can provide a good distraction and do some damage, they aren't really that useful. The true value of this ability comes with the fact that the Hive Head is also a reagent container, which stores up to 50 units of whatever you want to put inside. While you can't get what you put into the Hive Head out in any way, summoned bees will pull from the reagents inside your Hive Head to inject into targets, increasing the amount of !!FUN!! you can have with them drastically. For example, if you go into the bar and dump a bunch of different drinks into the Hive Head, your summoned bees will have equal chance to select any of those drinks to be what they inject into people. Reagents inside the Hive Head will stay there until it is dispelled, which upon re-using the ability, the Hive Head will be empty.

 Hive Head also has the additional utility that Release Bees works at any time so long as you're still wearing it. However, only 1 bee will be summoned whilst you are in hard crit or dead. Release Bees has a cooldown of 30 seconds.

Additionally, if mixing chemical cocktails to have a swarm of bees forcefully inject into people isn't to your liking, Hive Head provides an alternate mode of use if a legion core is used on it. Using a legion core on the Hive Head will cause it to mutate, changing its appearance and functionality. The Legion Hive Head, as opposed to releasing 6 bees, releases 4 legion skulls to fight for you instead. These legion skulls do more damage than bees, but can be dispelled by any attack and only last for 10 seconds as opposed to 20, but this also comes with the added benefit of a shorter cooldown time of 15 seconds as opposed to 30. The major benefit of this version of the ability is that if the skulls down a humanoid target, the skulls will turn that person into an allied legion, who will use their abilities to support you with healing and additional skull summoning should they run into a target. Just as disabling Hive Head normally removes chemicals, Hive Head will return to normal upon deactivating and reactivating the ability, requiring another legion core be used to access this version again.

image

Additionally, this PR adds the sound of a swarm of bees for when the ability gets activated, new sprites to facilitate implementing this ability, and some changes to how the armor subset of changeling abilities work internally in order to add ones that only apply a helmet or armor as opposed to both. This PR also fixes a bug with bees having an incorrect sprite offset (they look like they're one tile over to the left currently).

As always, the numbers and mechanics can be changed if need be, and I'm open to hearing others' thoughts about this.

## Why It's Good For The Game

Changeling hasn't received a new ability in a very long time, only reworks to their existing kit, so there haven't been any new toys to shake up the experience with. Hive Head is something new that fits Changeling's preference for one-on-one combat (the bees pretty much always aggro on the same target) while also playing into that weird alien shapeshifter vibe. Along with this, being able to change what chemicals the bees inject into people opens the door to creative ways of using it alongside the rest of the sandbox, so it'll be interesting to see what players come up with. The additional legion mode is just a cherry on top, allowing people to play around with the cool legion skull mechanics in an antagonistic setting.

## Changelog

:cl:
add: New Changeling Ability: Hive Head
fix: Fixed bees having an improper sprite offset
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
Amazing concept.

1 bee being released in hardcrit is too little IMO. I appreciate the detail but at only one release it's rather pointless. 2-3 would be more interesting.

**Comment by Time-Green** [comment]
Super cool idea! Although I worry if maybe the legion hive will have too much round-ending potential (or at the very least a Livrah video)

**Comment by Jacquerel** [comment]
Just as visual polish, it looks a little bad if the bees are simply deleted and vanish into thin air.  
Could you make a bee subtype with a timed life instead, in order to give it a similar "death animation" to the legion skulls?

**Comment by Shadowflame909** [comment]
Coroner Notta-Ling: Hey mining can I have a legion core? No reason specifically...

Sounds like fun in maints!

**Comment by Pickle-Coding** [comment]
Is there any way someone could find out that sticking a legion core would transform it?

Also the reagent the bee chooses should be weighted by the volume of the reagent.

**Comment by Admiral-Gigaton** [comment]
Infinitely based PR

**Comment by CCC23** [comment]
finally I can now become dr legion

**Comment by SomeoneYouProbablyKnow** [comment]
Based, but 6 beers with potentially instantly lethal injections is a bit much. Maybe only some of the bees should get the chem? 
3 out of 6? That way people will still get distracted by the murder hornets but getting stung is not opps dead all the time.

But anyway very much based, cant wait to have cargo ordering beekeeping suits to fight lings.

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -582,3 +587,123 @@
...
 	ADD_TRAIT(src, TRAIT_NODROP, CHANGELING_TRAIT)
+
+/datum/action/changeling/suit/hive_head
+	name = "Hive Head"
+	desc = "We coat our head in a waxy outing coating similar to a bee hive which can be used to manufacture bees to attack our enemies. Costs 15 chemicals."
+	helptext = "While the hive head does not provide much in the ways of armor, it does allow the user to send bees out to attack targets. Reagents can poured inside the hive to cause all bees released to inject said reagents."
+	button_icon_state = "hive_head"
+	chemical_cost = 15
+	dna_cost = 2
+	req_human = FALSE
```
Comment (Jacquerel) [inline code/modules/antagonists/changeling/powers/mutations.dm:598]: Double check how this combos with monkey ventcrawling, don't allow someone in pipes to spawn bees or legion skulls "from thin air" while in a pipe.

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 2]
```diff
@@ -582,3 +587,123 @@
...
+	energy = 10
+	bio = 50
+
+/obj/item/clothing/head/helmet/changeling_hivehead/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, CHANGELING_TRAIT)
+
+/obj/item/clothing/head/helmet/changeling_hivehead/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(istype(attacking_item, /obj/item/organ/internal/monster_core/regenerative_core/legion) && holds_reagents)
```
Comment (Jacquerel) [inline code/modules/antagonists/changeling/powers/mutations.dm:635]: Invert this for an early return, doesn't seem like we do anything in any other case.

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 3]
```diff
@@ -582,3 +587,123 @@
...
+		wearer.temporarilyRemoveItemFromInventory(wearer.head, TRUE)
+		wearer.equip_to_slot_if_possible(new /obj/item/clothing/head/helmet/changeling_hivehead/legion(wearer), ITEM_SLOT_HEAD, 1, 1, 1)
+		qdel(attacking_item)
+
+
+/datum/action/cooldown/hivehead_spawn_minions
+	name = "Release Bees"
+	desc = "Release a group of bees to attack all other lifeforms."
+	background_icon_state = "bg_demon"
+	overlay_icon_state = "bg_demon_border"
```
Comment (Jacquerel) [inline code/modules/antagonists/changeling/powers/mutations.dm:648]: Are these the icon states other changeling abilities use?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by Admiral-Gigaton** [comment]
It consumes chems from the thing, right? If I get only 1 unit of like adminzine it won't infinitely produce it?

**Comment by IndieanaJones** [comment]
> It consumes chems from the thing, right? If I get only 1 unit of like adminzine it won't infinitely produce it?

Not currently! Though if you deactivate the Hive Head, all chemicals currently inside it go away

**Review by Wallemations — APPROVED** [review]

File: code/modules/antagonists/changeling/powers/mutations.dm (hunk 3 above)
Comment (IndieanaJones) [reply code/modules/antagonists/changeling/powers/mutations.dm:648]: No, but seeing how this ability isn't tied to the rest of changeling's systems (no chem cost, cooldown-based ability), I figured having a different background would help make that apparent.

**Comment by IndieanaJones** [comment]
> This would also allow you to make them qdel on death and play the same animation, instead of leaving a corpse which sticks around until the other bees despawn, which I assume is the current behaviour?

Currently, the bee do vanish once 20 seconds is reached. However, if the bee is killed, the corpse stays because bees are one of those mobs that create an item to represent their dead corpse while they themselves get deleted. Note that there is a bee subtype which kills itself currently after 50(?) seconds, but they also leave their corpses behind. Not sure what the best choice of action is there.

**Comment by moocowswag** [comment]
Is the cooldown for bee/legion release per body or a cooldown that is stored in the antag itself (chemicals) if its the former it would allow spam with slime person

**Comment by Jacquerel** [comment]
> > This would also allow you to make them qdel on death and play the same animation, instead of leaving a corpse which sticks around until the other bees despawn, which I assume is the current behaviour?
> 
> Currently, the bee do vanish once 20 seconds is reached. However, if the bee is killed, the corpse stays because bees are one of those mobs that create an item to represent their dead corpse while they themselves get deleted. Note that there is a bee subtype which kills itself currently after 50(?) seconds, but they also leave their corpses behind. Not sure what the best choice of action is there.

I think that making this spawn a timed-life bee subtype which spawns a death animation upon death instead of a corpse item would solve both of these problems.

File: code/modules/antagonists/changeling/powers/mutations.dm (hunk 3 above)
Comment (Jacquerel) [reply code/modules/antagonists/changeling/powers/mutations.dm:649]: I personally feel like having them fit with other changeling abilities would look better, but it's you call.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**